src/github/api.rs

```diff
@@ -57,22 +58,155 @@ pub trait Session: Send + Sync {
     fn get_timeline(&self, owner: &str, repo: &str, number: u32) -> Result<Vec<TimelineEvent>>;
 }
 
+
+pub trait GithubSessionFactory: Send + Sync {
```

Comment:
not super happy w/ the naming here `/shrug`.  Open to suggestions if you have any. :slightly_smiling_face: